### PR TITLE
[Benchmark] Add kv_cache PCC check

### LINKED ISSUE #4783: Enable KV cache PCC check for benchmark LLMs
Currently in benchmarks PCC is calculated for prefill and first decode logits (for User 0).

Next step should be enabling KV cache PCC check.

### PR DESCRIPTION
**Pull request by vkovacevicTT**
### Ticket
Closes #4783 

### Problem description
Currently we measure PCC only for first user's prefill and first decode logits.
As a next step in extending PCC check, we want to add PCC check for KV cache.

### What's changed
- Added KV cache PCC check, for same number of steps as output logits PCC check.
- Modified `generate_and_benchmark` teacher-forcing logic so it can teacher-force a smaller window of ground truth tokens, and then continue autoregressivelly to full length of decode steps. This removes the need to split prefill and decode calls introduced in https://github.com/tenstorrent/tt-xla/commit/392e200b9ff69edcea01568330ed1ac09ca8b016

### Notes
We expect many models to fail because of this issue: #4785 

### Checklist
- [ ] [n150 LLMs](https://github.com/tenstorrent/tt-xla/actions/runs/26224737758)
- [ ] [LLMBOX LLMs](https://github.com/tenstorrent/tt-xla/actions/runs/26224756956)
- [ ] [Galaxy LLMs](https://github.com/tenstorrent/tt-xla/actions/runs/26224777626)
- [ ] [n150 LLMs on Marina's fix](https://github.com/tenstorrent/tt-xla/actions/runs/26505643781)
- [ ] [falcon3_10b_tp on Marina's fix](https://github.com/tenstorrent/tt-xla/actions/runs/26505570056)

### CODE REVIEW

File: tests/benchmark/benchmarks/llm_benchmark.py [hunk 1]
```diff
@@ -417,18 +425,30 @@ def benchmark_llm_torch_xla(
...
-            cpu_wrapper,
-            input_args,
-            torch.device("cpu"),
-            1,
-            verbose=False,
-            collect_logits=True,
-        )
+        # Run NUM_PCC_DECODE_STEPS decode steps.
+        cpu_decode_logits: list = []
+        if NUM_PCC_DECODE_STEPS > 0:
```
Comment (odjuricicTT) [inline tests/benchmark/benchmarks/llm_benchmark.py]: This is always true? Is there a need for this if?

File: tests/benchmark/benchmarks/llm_benchmark.py [hunk 2]
```diff
@@ -417,18 +425,30 @@ def benchmark_llm_torch_xla(
...
+                NUM_PCC_DECODE_STEPS,
+                verbose=False,
+                collect_logits=True,
+            )
 
         cpu_output_logits = cpu_prefill_logits + cpu_decode_logits
 
+        # CPU output tokens, used to teacher-force the device PCC run
+        # for the first len(cpu_output_logits) steps.
+        cpu_output_tokens = torch.tensor(
```
Comment (odjuricicTT) [inline tests/benchmark/benchmarks/llm_benchmark.py:457]: Can we take the tokens directly from the wrapper instead of doing argmax here again?

File: tests/benchmark/benchmarks/llm_benchmark.py [hunk 3]
```diff
@@ -791,35 +781,67 @@ def benchmark_llm_torch_xla(
...
-            decode_pcc_value >= required_pcc
-        ), f"First decode PCC failed. PCC={decode_pcc_value:.6f}, Required={required_pcc}"
-        print(
-            "First decode PCC verification passed with PCC={:.6f}".format(
-                decode_pcc_value
+
+        # Pre-transfer device KV cache to CPU once.
+        cpu_pkv = decode_only_cache
+        device_layers_cpu = [
+            (layer.keys.to("cpu"), layer.values.to("cpu"))
```
Comment (odjuricicTT) [inline tests/benchmark/benchmarks/llm_benchmark.py]: Have you tried obtaining the kv cache by adding it to model output? It already is part of the output of the ttnn ir graph so it should not change the behaviour. Tho we need to test whether it impacts perf or memory usage.

It would be a more general approach instead of reading it from model layer data.

**Review by odjuricicTT — COMMENTED** [review]
What is the number of models failing? You should open an issue to track all models that fail this check so we have an overview.

File: tests/benchmark/benchmarks/llm_benchmark.py (hunk 2 above)
Comment (vkovacevicTT) [reply tests/benchmark/benchmarks/llm_benchmark.py:457]: Sure.

File: tests/benchmark/benchmarks/llm_benchmark.py (hunk 3 above)
Comment (vkovacevicTT) [reply tests/benchmark/benchmarks/llm_benchmark.py]: Yes, we could extract what we need from output.
